src/test/java/org/gridsuite/casevalidation/server/CaseValidationTest.java

```diff
@@ -48,54 +49,68 @@
+    public void testUnfoundNetwork() throws Exception {
+        UUID notFoundNetworkId = UUID.fromString("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa");
+
+        // network not existing
+        given(loadFlowService.run(eq(notFoundNetworkId), any())).willThrow(new PowsyblException());
+        mvc.perform(put("/v1/networks/{networkUuid}/validate", notFoundNetworkId))
+                .andExpect(result -> assertTrue(result.getResolvedException() instanceof PowsyblException));
+    }
+
     @Test
     public void test() throws Exception {
         UUID testNetworkId = UUID.fromString("7928181c-7977-4592-ba19-88027e4254e4");
-        UUID notFoundNetworkId = UUID.fromString("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa");
 
-        given(networkStoreService.getNetwork(testNetworkId, PreloadingStrategy.COLLECTION)).willReturn(createNetwork());
-        given(networkStoreService.getNetwork(notFoundNetworkId, PreloadingStrategy.COLLECTION)).willThrow(new PowsyblException());
+        //Loadlow converges with default parameters
+        ObjectMapper om = new ObjectMapper();
 
-        // network not existing
-        mvc.perform(put("/v1/networks/{networkUuid}/validate", notFoundNetworkId))
-                .andExpect(status().isNotFound());
+        List<LoadFlowResult.ComponentResult> componentResults = Collections.singletonList(new LoadFlowResultImpl.ComponentResultImpl(0, LoadFlowResult.ComponentResult.Status.CONVERGED, 5, "slackBusId", 0));
+        given(loadFlowService.run(eq(testNetworkId), argThat(params -> params.isTransformerVoltageControlOn()))).willReturn(new LoadFlowResultImpl(true, Collections.emptyMap(), null, componentResults));
 
-        // load flow without parameters (default parameters are used)
         MvcResult result = mvc.perform(put("/v1/networks/{networkUuid}/validate", testNetworkId))
                 .andExpect(status().isOk())
                 .andExpect(content().contentType(MediaType.APPLICATION_JSON))
                 .andReturn();
-        assertTrue(result.getResponse().getContentAsString().contains("status\":\"CONVERGED\""));
-
-        // load flow with parameters
-        LoadFlowParameters params = new LoadFlowParameters()
-                .setNoGeneratorReactiveLimits(true)
-                .setDistributedSlack(false);
-        OpenLoadFlowParameters parametersExt = new OpenLoadFlowParameters()
-                .setSlackBusSelector(new MostMeshedSlackBusSelector());
-        params.addExtension(OpenLoadFlowParameters.class, parametersExt);
-
-        ByteArrayOutputStream stream = new ByteArrayOutputStream();
-        JsonLoadFlowParameters.write(params, stream);
-        String paramsString = new String(stream.toByteArray());
-
-        result = mvc.perform(put("/v1/networks/{networkUuid}/validate", testNetworkId)
-                .contentType(MediaType.APPLICATION_JSON)
-                .content(paramsString))
+        JsonNode validationOk = om.readTree(result.getResponse().getContentAsString()).path("validationOk");
```

Comment:
You don't need t use an object mapper, you can directly assert with a andExpect 
https://mkyong.com/spring-boot/spring-test-how-to-test-a-json-array-in-jsonpath/